feat(Recruiters): form recruiters

YeLnatSs opened the pull request:
Recruiters form created
Image added on banner

YeLnatSs committed:
feat(Recruiters): form recruiters

abruere commented on src/i18n/source/pages.yaml:
```diff
@@ -107,6 +107,51 @@ pages:
+      fr: De belles images font la différence…
+    picture_incentive__EDITOR_LABEL:
+      en: "Incentive to upload attractive images"
+      fr: "Incitation à présenter des images de qualité"
```
I would move this to https://github.com/abruere/workingoo/blob/dev/src/i18n/theme/pages.yaml to ease updates and sync from upstream project in `i18n/source` files.

YeLnatSs replied:
Okay, I moved it to the other location.

abruere commented on src/i18n/source/prompt.yaml:
```diff
@@ -99,6 +99,9 @@ prompt:
     context: Used for any type of resource (contact, asset, content…)
+  ad_button:
+    en: Post the ad
+    fr: Poster l'annonce
```
Same.

YeLnatSs replied:
This one too.

abruere commented on src/layouts/HeroTwoColLayout.vue:
```diff
@@ -0,0 +1,41 @@
+          </div>
+          <div class="col-md-6">
+            <div class="hero__background bg-primary-gradient" />
+            <div class="hero__content text-center text-white">
```
I suggest you reuse existing `BasicHeroLayout.vue` since there is a slot to easily customize content.

https://github.com/abruere/workingoo/blob/e89998f6ed2ecfd52237da702b2eb2d25d304fe3/src/layouts/BasicHeroLayout.vue#L3-L8

You can add an outer slot in `BasicHeroLayout.vue` if you need more customization

```vue
<section class="hero hero__background-container"> 
   <div class="hero__background bg-primary-gradient" /> 
   <slot name="heroContentContainer">
     <!-- This is default content of heroContentContainer slot -->
     <div class="hero__content text-center text-white"> 
       <slot name="heroContent" /> 
     </div> 
  </slot>
</section> 
```

Note: you can only use _either_ `heroContent` or `heroContentContainer`, it’s all what you need to fully customize the content and keep current hero background.

YeLnatSs replied:
I worked from your changes and reuse `BasicHeroLayout.vue`. Based on `this.$route` element is displayed.

abruere commented on src/pages/Recruiters.vue:
```diff
@@ -0,0 +1,595 @@
+
+import PageComponentMixin from 'src/mixins/pageComponent'
+
+export default {
```
We won’t be able to maintain duplicates of pages.

I suggest you use the same `NewAsset.vue` page you can also restyle if needed (depending on `this.$route`)

YeLnatSs replied:
No problem. Thank you

abruere commented on src/router/routes.js:
```diff
@@ -44,6 +44,12 @@ export const defaultRoutes = {
+      path: 'r',
+      component: () => import(/* webpackChunkName: 'landing' */ 'pages/Recruiters.vue'),
+      meta: {}
+    },
```
Here you can also map to newAsset, and detect `this.$route.name` in this page to know if it `newAsset` or `recruiters`, to reuse the same component and avoid duplicating code.

```
{
      name: 'recruiters',
      path: 'r',
      component: () => import(/* webpackChunkName: 'landing' */ 'pages/NewAsset.vue'),
      meta: {}
    },
```

YeLnatSs replied:
I did Thank you

abruere reviewed (commented):
Thanks for contributing!

As explained below we need to reuse existing code as much as possible to prevent bugs in the future.

The page looks very much like current `NewAsset`, except for the hero that can be customized. Same for input style.

https://deploy-preview-1--stelace-instant-workingoo-test.netlify.com/r
https://deploy-preview-1--stelace-instant-workingoo-test.netlify.com/n


Thanks to this PR I also spotted a responsive bug in `NewAsset.vue` with `title` input I’ll fix from upstream jobs project.

abruere commented:
I’ve fixed the input and made changes to `BasicHeroLayout.vue` I suggested above https://github.com/stelace/jobs-marketplace-template/commit/0ca95c981b14cddf8b2e7ff9ff48f33b351371fe to ease your work. 

@YeLnatSs you can rebase on dev branch to use this updated component instead of `HeroTwoColLayout.vue`, and get other latest updates in your fork.

```
git checkout dev
git fetch upstream
git merge upstream/dev
git checkout form-recruiters
git rebase dev
```

YeLnatSs committed:
Merge remote-tracking branch 'upstream/dev' into dev
 merge correction

YeLnatSs committed:
feat(Recruiters): Edit on recruiters page+ delete file: Recruiters.vue and HeroTwoColLayout

abruere commented on src/pages/NewAsset.vue:
```diff
@@ -121,39 +122,6 @@ export default {
-
-      // Index of first falsy step
-      return steps.indexOf(false) >= 0 ? steps.indexOf(false) - 1 : steps.length - 1
-    },
```
Are sure you want to get rid of several steps revealed as user fills the form?

Antoine specifically asked for this initially but he may have changed his mind.

You can disable this behavior only when `isRecruiters === true` if you want to keep steps for other users.

```js
  step () {
    const steps = [
      true, // fictive step 0
      true,
      false,
      false,
      false,
    ]
    if (isRecruiters) return steps.map(s => true)

    if (this.name.length >= 1) { // with high debounce to reduce distraction while typing
      steps[2] = true
    }
    …
```

YeLnatSs replied:
Miguel told me to remove the effect that reveals the fields once filled because it was causing problems and for UX design.

abruere reviewed (commented):
Thanks for the changes and reusing existing components.

Please find a question about NewAsset form below.

If you’re satisfied with the preview (link in netlify CD details below), you can rebase your branch and let me know so I can merge into dev branch.

YeLnatSs committed:
Merge remote-tracking branch 'upstream/dev' into dev
new merge dev

YeLnatSs commented:
@abruere It seems good to me. I rebased my branch

YeLnatSs committed:
feat(Recruiters): banner design+ title color and input wording changed

YeLnatSs committed:
feat(Recruiters): edit on recruiters form list and mobile

abruere reviewed (changes requested):
Hi @YeLnatSs, this PR currently breaks header:

![deploy-preview-1--stelace-instant-workingoo-test netlify com](https://user-images.githubusercontent.com/12909094/62572753-c61d7280-b894-11e9-84d4-6215cee3fe77.png)

abruere commented on src/layouts/MainLayoutHeader.vue:
```diff
@@ -355,6 +358,16 @@ export default {
+          v-show="isRecruiters"
+          :class="{ 'recruiters-title': isRecruiters }"
+          class="text-warning text-weight-medium text-center text-uppercase text-subtitle1"
+        >
```
Broken header is probably due to your `v-show` being on the child of the div you added, rather than on parent:

```js
<div class="q-mx-md col-md-9">
        <div
          v-show="isRecruiters"
          :class="{ 'recruiters-title': isRecruiters }"
          class="text-warning text-weight-medium text-center text-uppercase text-subtitle1"
        >
        {{ $t({ id: 'pages.recruiters.tagline' }) }}
       </div>
</div>
```

to 

```js
<div v-show="isRecruiters" class="q-mx-md col-md-9">
        <div
          :class="{ 'recruiters-title': isRecruiters }"
          class="text-warning text-weight-medium text-center text-uppercase text-subtitle1"
        >
        {{ $t({ id: 'pages.recruiters.tagline' }) }}
       </div>
</div>
```

YeLnatSs replied:
Okay, I changed it. I'm waiting for a feedback before commit it

YeLnatSs committed:
feat(Recruiters): mobile version+ Fix bug on header

YeLnatSs committed:
feat(Recruiters): edit design form

YeLnatSs committed:
feat(Recruiters): mobile spacing and bg input

YeLnatSs committed:
feat(Recruiters): edit padding and text

YeLnatSs committed:
feat(Recruiters): edit text

abruere commented on src/i18n/source/asset.yaml:
```diff
@@ -164,5 +164,5 @@ asset:
     quantity_label:
-      en: Number of people to be filled
-      fr: Nombre de personnes à pourvoir
+      en: Number of job to be filled
```
typo: `jobs`

YeLnatSs replied:
Thanks. I changed it

abruere commented on src/i18n/theme/pricing.yaml:
```diff
@@ -36,5 +36,5 @@ pricing:
     price_label:
-      en: Gross salary
+      en: Net salary
       fr: Salaire brut
```
EDIT: ~If changed in `en` it also needs to be changed to `Salaire net` in `fr`.~

Done in next commit thanks :)

abruere reviewed (approved):
Looks ready for merge once you’ve resolved conflicts due to recent updates.

I also suggest you add some padding to `Tout commence par un bon titre` line before first input on /n reworked page.

_Note: I let you check with Workingoo that `/r` URL for this new page is ok._

YeLnatSs committed:
feat(Recruiters): padding on title and edit text

YeLnatSs committed:
Merge branch 'dev' into form-recruiters

YeLnatSs commented:
> Looks ready for merge once you’ve resolved conflicts due to recent updates.
> 
> I also suggest you add some padding to `Tout commence par un bon titre` line before first input on /n reworked page.
> 
> _Note: I let you check with Workingoo that `/r` URL for this new page is ok._

Thank. I fixed conflicts and I added some padding on `Tout commence par un bon titre`

abruere commented:
Netlify reports some [linting errors](https://app.netlify.com/sites/stelace-instant-workingoo-test/deploys/5d514d9aaa320700099ecb01). Can you fix them before I merge?

YeLnatSs committed:
feat(Recruiters): fix lint

YeLnatSs commented:
> Netlify reports some [linting errors](https://app.netlify.com/sites/stelace-instant-workingoo-test/deploys/5d514d9aaa320700099ecb01). Can you fix them before I merge?

I fixed them. Normally, you won't have problem to merge.
